modules/ml/src/main/java/org/apache/ignite/ml/selection/cv/CrossValidationResult.java

```diff
@@ -110,21 +107,21 @@ void setBestHyperParams(Map<String, Double> bestHyperParams) {
      *
      * @return The Scoring Board.
      */
-    public Map<Map<String, Double>, double[]> getScoringBoard() {
-        return scoringBoard;
+    public synchronized Map<Map<String, Double>, double[]> getScoringBoard() {
+        return Collections.unmodifiableMap(scoringBoard);
```

Comment:
I agree that `synchronized` is not the best choice for this functionality, but it's OK. 

At the same time, this exact place is not "OK", we return the map from this place and after it's returned in the outer world it's not guarded anymore by `synchronized`.